app/scripts/migrations/097.ts

```diff
@@ -0,0 +1,49 @@
+import { cloneDeep, isEmpty } from 'lodash';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 97;
+
+/**
+ * Remove `nonceDetail` from transactions
+ *
+ * @param originalVersionedData
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, any>) {
+  const TransactionController = state?.TransactionController || {};
+  const transactions = state?.TransactionController?.transactions || {};
```

Comment:
To avoid some minor duplication, should we reference the above variable here so `transactionControllerState.transactions || {}`?

Reply:
Done

Reply:
Done